base/MaplGrid.F90

```diff
@@ -325,6 +329,21 @@ subroutine MAPL_GridGet(GRID, globalCellCountPerDim, localCellCountPerDim, RC)
          end if
       end if
 
+      if (present(layout)) then
+         call ESMF_GridGet(grid,distgrid=distgrid,dimCount=dimCount,_RC)
+         call ESMF_DistGridGet(distgrid,deCount=deCount,_RC)
+         allocate(minindex(dimCount,decount),maxindex(dimCount,decount))
+
+         call MAPL_DistGridGet(distgrid, &
+            minIndex=minindex, &
+            maxIndex=maxindex, _RC)
+         call MAPL_GetImsJms(Imins=minindex(1,:),Imaxs=maxindex(1,:),&
+           Jmins=minindex(2,:),Jmaxs=maxindex(2,:),Ims=ims,Jms=jms,_RC)
+
+         layout(1) = size(ims)
+         layout(2) = size(jms)
+      end if
+
```

Comment:
I think we should deallocate ims and jms at this point

Reply:
@atrayano yep, made that change